File: app/src/main/java/com/mapbox/mapboxsdk/plugins/testapp/PluginApplication.java
package com.mapbox.mapboxsdk.plugins.testapp;

import android.app.Application;

import com.mapbox.mapboxsdk.Mapbox;
import com.squareup.leakcanary.LeakCanary;

import timber.log.Timber;

public class PluginApplication extends Application {

  @Override
  public void onCreate() {
    super.onCreate();
    if (LeakCanary.isInAnalyzerProcess(this)) {
      return;
    }

    LeakCanary.install(this);
    initializeLogger();
    Mapbox.getInstance(this, getString(R.string.mapbox_access_token));
  }

  private void initializeLogger() {
    if (BuildConfig.DEBUG) {
      Timber.plant(new Timber.DebugTree());
    }
  }
}

File: app/src/main/java/com/mapbox/mapboxsdk/plugins/testapp/PluginApplication.kt
package com.mapbox.mapboxsdk.plugins.testapp

import androidx.multidex.MultiDexApplication

import com.mapbox.mapboxsdk.Mapbox

import timber.log.Timber

class PluginApplication : MultiDexApplication() {

    override fun onCreate() {
        super.onCreate()
        initializeLogger()
        Mapbox.getInstance(this, getString(R.string.mapbox_access_token))
    }

    private fun initializeLogger() {
        if (BuildConfig.DEBUG) {
            Timber.plant(Timber.DebugTree())
        }
    }
}
